### PR TITLE
set structure() as an undesirable function by default

### PR DESCRIPTION
**Pull request by MichaelChirico**
Follow-up to #2227

### CODE REVIEW

File: R/zzz.R [hunk 1]
```diff
@@ -194,12 +196,14 @@ default_undesirable_functions <- all_undesirable_functions[names(all_undesirable
...
   "source",
+  "structure",
   "Sys.setenv",
   "Sys.setlocale",
   "Sys.unsetenv",
   "trace",
   "undebug",
-  "untrace"
+  "untrace",
+  NULL
```
[the remark this answers is not in the document]
Comment (MichaelChirico) [reply R/zzz.R:206]: it will shrink any future diff from appending to the end of the list (instead of 2 lines diff, one of which is just adding a trailing comma, the diff is focused on the one line added). Similar motivation in Python (c.f. `[1,2,]` and `(1,2,)`) and Go for allowing trailing `,` in constructors (in fact it's _required_ in Go: https://dave.cheney.net/2014/10/04/that-trailing-comma).

Also here:

https://github.com/r-lib/lintr/blob/8d420b5a035645116be2e1ad216ebea5cabe4ce5/R/shared_constants.R#L159

Basically accomplishing `rlang::list2()` for `c()`

**Comment by AshesITR** [comment]
LGTM holding off merging until NEWS is up to speed so we can't forget the bullet.

File: .lintr [hunk 2]
```diff
@@ -31,7 +31,7 @@ linters: linters_with_defaults(
     sort_linter(),
     sprintf_linter(),
     strings_as_factors_linter(),
-    undesirable_function_linter(c(Sys.setenv = NA_character_, mapply = NA_character_)),
+    undesirable_function_linter(c(Sys.setenv = NA_character_, mapply = NA_character_, structure = NA_character_)),
```
Comment (AshesITR) [inline .lintr:34]: Didn't we clean this up?

Comment (MichaelChirico) [reply .lintr:34]: Not sure I follow...

Comment (AshesITR) [reply .lintr:34]: NVM I had thought we were using all default undesirables.

File: R/zzz.R [hunk 3]
```diff
@@ -171,7 +171,9 @@ all_undesirable_functions <- modify_defaults(
   untrace = paste(
     "remove this likely leftover from debugging.",
     "It is only useful for interactive debugging with trace()"
-  )
+  ),
+  structure =
```
Comment (AshesITR) [inline R/zzz.R:175]: Is it possible to do a trailing comma trick here, too?

Comment (MichaelChirico) [reply R/zzz.R:175]: not trivially -- rlang::list2() inside modify_defaults() would be the right approach to let someone else handle the implementation, or we'd have to hack it together ourselves looking for missing() at the end of ...

Comment (AshesITR) [reply R/zzz.R:175]: Okay, never mind then.

**Review by AshesITR — APPROVED** [review]
